better type resolution

jacoobes opened the pull request:
I was checking out the website you used for resolving typings. I made some changes to improve it even more and fix the masquerading esm warning 
https://arethetypeswrong.github.io
If you run npm pack, take the tgz, and use the same website, the resolution of typings seems to be better.

- Features 
    - Resolves rxjs-operators for node16, bundler
    - removes unneeded package.json in rxjs-operators directory

jstasiak commented:
Hey @jacoobes, thank you very much for this contribution.

Unfortunately we're unlikely to accept this PR in its current shape:

* It bundles multiple unrelated changes (platform independent cleanup, removal of a dependency, JS replacement of some POSIX tooling, typing improvement)
* The platform-independent code is a bit problematic on its own – it seems pretty verbose and we're likely to forget to keep things Windows-friendly as no one uses Windows on our side. It's an unfortunate situation to be in, I suggest using WSL to work around the problems on Windows.

I'll be more than happy to accept this change if it's just the typing improvement in this PR and the description explains what's improved/how/etc.

Thank you once again

jacoobes commented:
ah, understandable! 
I can revert the scripts and reinstall the dependency. Would it still be contributable?
I have linux on a separate laptop, so I can help develop this package on there.

jacoobes commented:
@jstasiak Hey, I reverted the scripts and cleaned up the copyfiles part, since i found out it wasnt needed.

Checking ts-results-es on npm shows that it failed for node16 ( commonjs users would have to dynamically import the rxjs-operator), but this should fix it. I also remove an unneeded package.json in rxjs-operators/, cleans up the build:copy script a little

jstasiak commented:
> I can revert the scripts and reinstall the dependency. Would it still be contributable?

Yes, absolutely!

> Checking ts-results-es on npm shows that it failed for node16 ( commonjs users would have to dynamically import the rxjs-operator), but this should fix it. 

That's actually already fixed, just not released yet: #53

jstasiak commented:
Yup, if you reduce the PR description (with the title) and the diff to only that I'll be happy to merge it.

jacoobes commented:
might redo this pr. not sure how to get rid of the large package.lock diff
